front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -69,12 +71,18 @@ const usedModelConfigs = [
 
 // Actions
 
-const ACTION_MODES = ["GENERIC", "RETRIEVAL", "DUST_APP_RUN"] as const;
+const ACTION_MODES = [
+  "GENERIC",
+  "RETRIEVAL_SEARCH",
+  "RETRIEVAL_EXHAUSTIVE",
+  "DUST_APP_RUN",
+] as const;
 type ActionMode = (typeof ACTION_MODES)[number];
 const ACTION_MODE_TO_LABEL: Record<ActionMode, string> = {
   GENERIC: "No action (Generic model)",
-  RETRIEVAL: "Search Data Sources",
-  DUST_APP_RUN: "Execute Dust App",
+  RETRIEVAL_SEARCH: "Search Data Sources",
+  RETRIEVAL_EXHAUSTIVE: "Exhaustive Data  Sources Processing",
```

Comment:
NIT: what do you think about this proposal of labels?

<img width="226" alt="Capture d’écran 2023-10-23 à 10 51 29" src="https://github.com/dust-tt/dust/assets/3803406/8a0a9124-d06d-4e71-b89a-e198d008cf70">

Suggestion rewording: 
- No retrieval of Data Sources (only the generic model)
- Retrieval of selected Data Sources based on semantic search
- Full retrieval of selected Data Sources 
- Execute a Dust app

Reply:
Thanks for the suggestions :pray: At the time I think the Figma's wording is better (shorter, less technical) although I agree it's not perfect. 
It's a matter of taste though, and if you rally @Duncid or @gabhubert, I'll change it in a follow-up PR

Reply:
I agree to disagree, it was a suggestion 👍🏻